tests/pkg_umorse/main.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2017 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ * @file
+ * @brief       Check if the uMorse package builds.
+ *
+ * @author      Sebastian Meiling <s@mlng.net>
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "board.h"
+#include "led.h"
+#include "periph_conf.h"
+#include "xtimer.h"
+#define UMORSE_MSLEEP(m)    xtimer_usleep(m * US_PER_MS)
+
+#include "umorse.h"
+#include "print.h"
+
+#define CODE_LEN	(64U)
+
+void blink_dit(void *args, uint8_t flags)
+{
+    (void) args;
+    (void) flags;
+
+    LED0_ON;
+    UMORSE_MSLEEP(UMORSE_DELAY_DIT);
+    LED0_OFF;
+}
+
+void blink_dah(void *args, uint8_t flags)
+{
+    (void) args;
+    (void) flags;
+
+    LED0_ON;
+    UMORSE_MSLEEP(UMORSE_DELAY_DAH);
+    LED0_OFF;
+}
+
+void blink_nil(void *args, uint8_t flags)
+{
+    (void) args;
+
+    uint8_t cnt = flags & UMORSE_MASK_COUNT;
```

Comment:
This seems off: ```flags & UMORSE_MASK_COUNT``` might yield up to 16. ```UMORSE_DELAY_DIT``` is defined to ```120``` in the makefile. But ```cnt``` is of type ```uint8_t```, which cannot hold the result of the multiplication.

Reply:
I didn't think about to be a problem because  `UMORSE_MSLEEP(cnt * UMORSE_DELAY_DIT)` will be resolved to `xtimer_usleep(cnt * UMORSE_DELAY_DIT)` which takes an `unint32_t` as input param. I thought that would fit, your comment suggests, that the multiplication will/could be _cut off_ at `UINT8_MAX` (255)? 

So it should read ` UMORSE_MSLEEP((uint32_t)cnt * UMORSE_DELAY_DIT)` to be safe?

Reply:
I tested, and you're right, the result is ok. I thought the result of a multiplication in C is always of the type of the leftmost multiplicant. Seems like for arguments, it is the argument type. Sorry for the noise.